src/main/java/de/hysky/skyblocker/utils/ItemUtils.java

```diff
@@ -206,13 +207,19 @@ public static PropertyMap propertyMapWithTexture(String textureValue) {
     public static String getHeadTexture(ItemStack stack) {
         if (!stack.isOf(Items.PLAYER_HEAD) || !stack.contains(DataComponentTypes.PROFILE)) return "";
 
-        ProfileComponent profile = stack.get(DataComponentTypes.PROFILE);
-        String texture = profile.properties().get("textures").stream()
-                .map(Property::value)
-                .findFirst()
-                .orElse("");
+        Iterator<Property> iterator = stack.get(DataComponentTypes.PROFILE)
+                                           .properties()
+                                           .get("textures")
+                                           .iterator();
 
-        return texture;
+        if (!iterator.hasNext()) return "";
+        return iterator.next().value();
+    }
+
+    public static Optional<String> getHeadTextureOptional(ItemStack stack) {
+        String texture = getHeadTexture(stack);
+        if (texture == null) return Optional.empty();
```

Comment:
The method can return a blank string if it's not there (and I don't know if it can return a null value?) so you would want to change this so that the optional represents its functionality